### PR TITLE
Readme env

### PR DESCRIPTION
**Pull request by monsieurleberre**
Just creating the draft PR to see the changes

### CODE REVIEW

File: src/Trakx.Common.Tests/Unit/Config/ConfigurationProviderTests.cs [hunk 1]
```diff
@@ -19,25 +17,93 @@ public ConfigurationProviderTests(ITestOutputHelper output)
...
+        // Note 3: As of now, the iteration over the vars names will insert each missing property on top of the list. It is possible to refine it to make sure some specfic sorting is applied.
         [Fact]
         public void UpdateReadmeTest()
         {
-            string codeBase = Assembly.GetExecutingAssembly().Location;
-            UriBuilder uri = new UriBuilder(codeBase);
-            string path = Uri.UnescapeDataString(uri.Path);
-            var directory = Path.GetDirectoryName(path);
+            string directoryName = GetAssemblyDirectoryName();
+            var readmeFilePath = directoryName + "/../../../../../readme.md";
```
Comment (monsieurleberre) [inline src/Trakx.Common.Tests/Unit/Config/ConfigurationProviderTests.cs:28]: Use Directory Info.Parent to make this less environment dependent :)

File: src/Trakx.Common.Tests/Unit/Config/ConfigurationProviderTests.cs [hunk 2]
```diff
@@ -19,25 +17,93 @@ public ConfigurationProviderTests(ITestOutputHelper output)
...
+            var readmeLines = GetReadmeLines(readmeFilePath);
 
+            var insertLineNumber = GetKeysInsertLineNumber(readmeLines);
+
+            var configProperties = GetConfigurationProperties();
+
+            configProperties
+                .ForEach(x =>
+                {
+                    var keyLine = $"-   {x}";
```
Comment (monsieurleberre) [inline src/Trakx.Common.Tests/Unit/Config/ConfigurationProviderTests.cs:39]: The little "-" will make it harder for Devs to copy paste the paragraph and build their .env file stub

File: src/Trakx.Common.Tests/Unit/Config/ConfigurationProviderTests.cs [hunk 3]
```diff
@@ -19,25 +17,93 @@ public ConfigurationProviderTests(ITestOutputHelper output)
             _output = output;
         }
 
+        // TODO: CS - Unit Test used to write the Env variables names into the Readme.md
+        // Note 1: Should test on a *nix style OS if the path detection works fine
+        // Note 2: The config and the attribute classes are here just as a placeholder. If the approach is OK, I will move them (or move to the right place)
+        // Note 3: As of now, the iteration over the vars names will insert each missing property on top of the list. It is possible to refine it to make sure some specfic sorting is applied.
```
Comment (monsieurleberre) [inline src/Trakx.Common.Tests/Unit/Config/ConfigurationProviderTests.cs:23]: Cool, no need to make it too complex, this is only a mini convenience Dev, so let's keep it easy

**Review by monsieurleberre — COMMENTED** [review]
Cool, let's make sure we inspect all referenced assemblies for those attributes, but maybe let's make it faster with a filter on the name of the class (they need to end by Configuration for instance).

**Comment by monsieurleberre** [comment]
This PR was just there for communicating about code changes, i think the branch has too many commits and should only include the commits you have written

File: src/Trakx.Common.Tests/Unit/Config/ConfigurationProviderTests.cs (hunk 2 above)
Comment (carlos-takeapps) [reply src/Trakx.Common.Tests/Unit/Config/ConfigurationProviderTests.cs:39]: the "-" is for the markdown. It creates the list of names as a list. The test already writes the readme.md that is in the root of this repo. Do we need to also be able to create the stub file or modify Readme.md files for other repositories?

File: src/Trakx.Common.Tests/Unit/Config/ConfigurationProviderTests.cs (hunk 1 above)
Comment (carlos-takeapps) [reply src/Trakx.Common.Tests/Unit/Config/ConfigurationProviderTests.cs:28]: Yes, OK. I think it will make it more OS independent.

**Pull request closed without merging**